core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -125,7 +135,7 @@ protected void configure() {
         queryParserService = injector.getInstance(IndexQueryParserService.class);
         MapperService mapperService = queryParserService.mapperService;
         //create some random type with some default field, those types will stick around for all of the subclasses
-        currentTypes = new String[randomIntBetween(0, 5)];
+        currentTypes = new String[randomIntBetween(1, 5)];
```

Comment:
I temporarily changed this to add at least one type plus mappings because so far in our test setup the FieldNamesFieldMapper which we now need to be non-null in ExistQueryBuilder is initiallized only when triggered by the mapperService.merge. Before there were null checks in the ExistQueryParser/Builder newFilter() method that now got removed on master. Still looking for other ways to change the test setup here.